java/ql/lib/semmle/code/java/frameworks/OkHttp.qll

```diff
@@ -0,0 +1,55 @@
+/**
+ * Provides classes and predicates for working with the OkHttp client.
+ */
+
+import java
+private import semmle.code.java.dataflow.ExternalFlow
+
+private class OkHttpOpenUrlSinks extends SinkModelCsv {
+  override predicate row(string row) {
+    row =
+      [
+        "okhttp3;Request;true;Request;;;Argument[0];open-url",
+        "okhttp3;Request$Builder;true;url;;;Argument[0];open-url"
+      ]
+  }
+}
+
+private class OKHttpSummaries extends SummaryModelCsv {
+  override predicate row(string row) {
+    row =
+      [
+        "okhttp3;HttpUrl;false;parse;;;Argument[0];ReturnValue;taint",
+        "okhttp3;HttpUrl;false;uri;;;Argument[-1];ReturnValue;taint",
+        "okhttp3;HttpUrl;false;url;;;Argument[-1];ReturnValue;taint",
+        "okhttp3;HttpUrl$Builder;false;addEncodedPathSegment;;;Argument[-1];ReturnValue;value",
+        "okhttp3;HttpUrl$Builder;false;addEncodedPathSegments;;;Argument[-1];ReturnValue;value",
+        "okhttp3;HttpUrl$Builder;false;addEncodedQueryParameter;;;Argument[-1];ReturnValue;value",
+        "okhttp3;HttpUrl$Builder;false;addPathSegment;;;Argument[-1];ReturnValue;value",
+        "okhttp3;HttpUrl$Builder;false;addPathSegments;;;Argument[-1];ReturnValue;value",
+        "okhttp3;HttpUrl$Builder;false;addQueryParameter;;;Argument[-1];ReturnValue;value",
+        "okhttp3;HttpUrl$Builder;false;build;;;Argument[-1];ReturnValue;taint",
+        "okhttp3;HttpUrl$Builder;false;encodedFragment;;;Argument[-1];ReturnValue;value",
+        "okhttp3;HttpUrl$Builder;false;encodedPassword;;;Argument[-1];ReturnValue;value",
+        "okhttp3;HttpUrl$Builder;false;encodedPath;;;Argument[-1];ReturnValue;value",
+        "okhttp3;HttpUrl$Builder;false;encodedQuery;;;Argument[-1];ReturnValue;value",
+        "okhttp3;HttpUrl$Builder;false;encodedUsername;;;Argument[-1];ReturnValue;value",
+        "okhttp3;HttpUrl$Builder;false;fragment;;;Argument[-1];ReturnValue;value",
+        "okhttp3;HttpUrl$Builder;false;fragment;;;Argument[-1];ReturnValue;value",
```

Comment:
Duplicate rows; should one be adding taint?

Reply:
This was a copy-paste mistake, thanks for spotting it. Let's discuss taint on the comment below.

```suggestion
        "okhttp3;HttpUrl$Builder;false;fragment;;;Argument[-1];ReturnValue;value",
```